### PR TITLE
Adding websocket sample for win32

### PR DESCRIPTION
**Pull request by jplafonta**

### CODE REVIEW

File: Samples/Win32WebSocket/main.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    async_queue_handle_t queue;
+    uint32_t sharedAsyncQueueId = 0;
+    CreateSharedAsyncQueue(
+        sharedAsyncQueueId,
+        AsyncQueueDispatchMode::AsyncQueueDispatchMode_ThreadPool,
+        AsyncQueueDispatchMode::AsyncQueueDispatchMode_ThreadPool,
+        &queue);
+
+    std::string url = "wss://echo.websocket.org";
+
```
Comment (lubeltra) [inline Samples/Win32WebSocket/main.cpp]: extra blank line

File: Samples/Win32WebSocket/main.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        HCGetWebSocketSendMessageResult(asyncBlock, &result);
+
+        printf_s("HCWebSocketSendMessage complete: %d, %d\n", result.errorCode, result.platformErrorCode);
+        delete asyncBlock;
+    };
+
+    std::string requestString = "This message should be echoed!";
+    printf_s("Calling HCWebSocketSend and waiting for response...\n");
+    hr = HCWebSocketSendMessageAsync(websocket, requestString.data(), asyncBlock);
+    WaitForSingleObject(g_eventHandle, INFINITE);
```
Comment (lubeltra) [inline Samples/Win32WebSocket/main.cpp]: you don't set this in the second callback?

File: Samples/Win32WebSocket/main.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    _In_ HCWebSocketCloseStatus closeStatus
+    )
+{
+    printf_s("Websocket closed!\n");
+    SetEvent(g_eventHandle);
+}
+
+int main()
+{
+    g_eventHandle = CreateEvent(nullptr, false, false, nullptr);
```
Comment (lubeltra) [inline Samples/Win32WebSocket/main.cpp]: is this an auto reset event?

File: Samples/Win32WebSocket/main.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    };
+
+    std::string requestString = "This message should be echoed!";
+    printf_s("Calling HCWebSocketSend and waiting for response...\n");
+    hr = HCWebSocketSendMessageAsync(websocket, requestString.data(), asyncBlock);
+    WaitForSingleObject(g_eventHandle, INFINITE);
+
+    printf_s("Calling HCWebSocketDisconnect...\n");
+    HCWebSocketDisconnect(websocket);
+    WaitForSingleObject(g_eventHandle, INFINITE);
```
Comment (lubeltra) [inline Samples/Win32WebSocket/main.cpp]: who is supposed to set the event here?

File: Samples/Win32WebSocket/main.cpp (hunk 2 above)
Comment (jplafonta) [reply Samples/Win32WebSocket/main.cpp]: It gets set when the response is received.  I will add another set/wait for the send being completed.

---
In reply to: [211452794](https://github.com/Microsoft/libHttpClient/pull/254#discussion_r211452794) [](ancestors = 211452794)

**Pull request merged**